app/controllers/admin/app_integrations_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Admin
+  class AppIntegrationsController < Admin::ApplicationController
```

Comment:
This is one of the most difficult models to name I've dealt with in a long time!

I had a conversation about the nuances of this model with @joshpuetz last Friday, and the reason why `MobileApp` is undesirable is that it would tighten up its scope to Mobile apps only. With Rpush we can deliver PNs to macOS apps (Safari relies on this strategy), browsers could be configured as targets (sites/[browser extensions](https://github.com/forem/forem-browser-extension) 👀 ), [Windows apps](https://github.com/rpush/rpush#windows-notification-service-windows-81-10-apps--phone--80) and possibly other platforms.

The original name `PushNotificationTarget` doesn't roll out of the tongue, isn't easy to understand and have been brainstorming some options for a while [with ideas shared by @rhymes too](https://github.com/forem/forem/pull/13304#discussion_r610466864). This name allows us to shelter other "App related features" like the [Runtime Banner for Mobile Deep Linking](https://github.com/forem/forem/pull/13190), i.e. the creator can choose from the App Integrations configured to launch a native app (mobile, macOS, etc).

I want to share a writeup to include the product team in this decision though, because I think it's important before committing to a name for a feature that will be in the Admin dashboard. Something else to keep in mind: This is a somewhat niche config that will be used by a smaller fraction of creators (Forem Cloud will support the defaults via ENV variables similar to how Pusher is working today). I hope to have this write up published by the end of my Monday to close the loop on this soon 🤞

Reply:
Thank you for taking the time to come up with the correct naming, as while writing I also thought that we need to keep "app integrations" separate from plugins to the Forem app and at first glance I thought about plugins because they too can be "integrations to the app" if the "app" is considered the running app. Sorry to add to the confusion :)

Reply:
Ah, that's helpful context @fdoxyz, thanks! I'm still not 100% convinced by `App`  because it could mean so many things. From what you describe `Client` or even `ForemClient` seems like it could work but I'm pretty sure that whatever we name this a class-level comment is probably in order.